src/components/Arping/Arping.tsx

```diff
@@ -0,0 +1,217 @@
+            setLoading(false);
+            // Allow Saving as the output is finalised
+            setAllowSave(true);
+            setHasSaved(false);
+        },
+        [handleProcessData] // Dependency on the handleProcessData callback
+    );
+
+    /**
+     * Function to handle form submission.
+     * Executes the Arping component with the provided form values.
+     * @param {FormValuesType} values - Form values containing Arping input.
+     */
+    const onSubmit = async (values: FormValuesType) => {
+        // Activate loading state to indicate ongoing process.
+        setLoading(true);
+        // Disallow saving until the tool's execution is complete
+        setAllowSave(false);
+        // Construct arguments for the Arping component command based on form input.
+        let args = [values.targetIP];
+
+        if (values.count) {
+            args.push("-c", values.count);
+        }
+
+        if (values.interface) {
+            args.push("-I", values.interface);
+        }
+
+        if (values.verbose) {
+            args.push("-v");
+        }
+        //Execute the Arping component via helper method and handle its output or potential errors.
+        CommandHelper.runCommandWithPkexec("arping", args, handleProcessData, handleProcessTermination)
+            .then(() => {
+                setLoading(false);
+            })
+            .catch((error) => {
+                setOutput(`Error: ${error.message}`);
+                setLoading(false);
+            });
+        setAllowSave(true);
+    };
+
+    /**
+     * handleSaveComplete: Callback to handle the completion of the file saving process.
+     * It updates the state by indicating that the file has been saved and deactivates the save button.
+     */
+    const handleSaveComplete = () => {
+        // Indicating that the file has saved which is passed
+        // back into SaveOutputToTextFile to inform the user
+        setHasSaved(true);
+        setAllowSave(false);
+    };
+
+    /**
+     * clearOutput: Callback function to clear the console output.
+     * It resets the state variable holding the output, thereby clearing the display.
+     */
+    const clearOutput = () => {
+        setOutput("");
+        setHasSaved(false);
+        setAllowSave(false);
+    };
+
+    // Render component
+    return (
+        <RenderComponent
+            title={title}
+            description={description}
+            steps={steps}
+            tutorial={tutorial}
+            sourceLink={sourceLink}
+        >
+            {!loadingModal && (
+                <InstallationModal
+                    isOpen={opened}
+                    setOpened={setOpened}
+                    feature_description={description}
+                    dependencies={dependencies}
+                />
+            )}
+            <form onSubmit={form.onSubmit(onSubmit)}>
+                <Stack>
+                    {LoadingOverlayAndCancelButton(loading, pid)}
```

Comment:
This needs to be the loading overlay for pkexec
```suggestion
                    {LoadingOverlayAndCancelButtonPkexec(loading, pid, handleProcessData, handleProcessTermination)}
```

Reply:
Fixed.